raiden/tests/utils/apitestcontext.py

```diff
@@ -34,13 +36,82 @@ def decode_response(response):
 class ApiTestContext():
 
     def __init__(self, reveal_timeout):
+        self.events = list()
         self.channels = []
         self.tokens = set()
         self.channel_schema = ChannelSchema()
         self.channel_list_schema = ChannelListSchema()
-        self.events_list_schema = EventsListSchema()
         self.reveal_timeout = reveal_timeout
 
+    def add_events(self, events):
+        self.events += events
+
+    def specify_token_for_channelnew(self, token_address):
+        """Since it's not part of the event but part of the querying and we
+        mock the interface we should check that the token address properly makes
+        it through the REST api"""
+        self.token_for_channelnew = address_decoder(token_address)
+
+    def specify_channel_for_events(self, channel_address):
+        """Since it's not part of the event but part of the querying and we
+        mock the interface we should check that the channel address properly
+        makes it through the REST api"""
+        self.channel_for_events = address_decoder(channel_address)
+
+    def get_token_added_events(self, from_block, to_block):
+        return_list = list()
+        for event in self.events:
+            if (
+                    event['_event_type'] == 'TokenAdded' and
+                    event['block_number'] >= from_block and
+                    event['block_number'] <= to_block
```

Comment:
Nitpicking: I would prefer a test on the form:

```
is_valid_event = (
   ... multiline test
)

if is_valid_event:
    ... whatnot
```

Because I find multiline `if` statementsa little bit harder to read